### PR TITLE
[SYCLomatic] Change io_iterator_pair accessor naming and add do_swap_iters argument

### PR DESCRIPTION
**Pull request by jzc**
Signed-off-by: Cai, Justin [justin.cai@intel.com](mailto:justin.cai@intel.com)

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc [hunk 1]
```diff
@@ -1633,13 +1633,18 @@ sort_pairs(_ExecutionPolicy &&policy, key_t keys_in, key_out_t keys_out,
 template <typename _ExecutionPolicy, typename key_t, typename value_t>
 inline void sort_pairs(
     _ExecutionPolicy &&policy, io_iterator_pair<key_t> &keys,
-    io_iterator_pair<value_t> &values, int64_t n, bool descending,
+    io_iterator_pair<value_t> &values, int64_t n, bool descending = false,
```
Comment (0x12CC) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1636]: Why add a default value for `bool descending`?

Comment (jzc) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1636]: The other overloads have default values for descending, although I think this one was missed. See https://github.com/oneapi-src/SYCLomatic/pull/468#discussion_r1069612437

Comment (jzc) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1636]: @danhoeflinger confirm?

Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1636]: Yes, you are correct.

**Review by danhoeflinger — COMMENTED** [review]
`segmented_sort_pairs` needs the same adjustment for `.input()` and `.output()` as well as the swapping code.

**Review by 0x12CC — APPROVED** [review]

**Review by danhoeflinger — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc [hunk 2]
```diff
@@ -1774,13 +1774,17 @@ inline void segmented_sort_pairs(
     _ExecutionPolicy &&policy, io_iterator_pair<key_t> &keys,
     io_iterator_pair<value_t> &values, int64_t n, int64_t nsegments,
     OffsetIteratorT begin_offsets, OffsetIteratorT end_offsets,
-    bool descending = false, int begin_bit = 0,
+    bool descending = false, bool do_swap_iters = false, int begin_bit = 0,
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1777]: Seems a ill-formed syntax, please define variables in sperate line.

File: clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h [hunk 3]
```diff
@@ -1483,13 +1483,17 @@ inline void segmented_sort_pairs(
     _ExecutionPolicy &&policy, io_iterator_pair<key_t> &keys,
     io_iterator_pair<value_t> &values, int64_t n, int64_t nsegments,
     OffsetIteratorT begin_offsets, OffsetIteratorT end_offsets,
-    bool descending = false, int begin_bit = 0,
+    bool descending = false, bool do_swap_iters = false, int begin_bit = 0,
```
Comment (yihwang-nv) [inline clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h:1486]: Seems a ill-formed syntax, please define variables in sperate line.

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc (hunk 2 above)
Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1777]: These are all default parameters in a parameter list, so I'm not sure the problem here.

File: clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h (hunk 3 above)
Comment (danhoeflinger) [reply clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h:1486]: Same as above, default parameters in a function's parameter list.

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc (hunk 2 above)
Comment (yihwang-nv) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc:1777]: Ah, my mistake.

File: clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h (hunk 3 above)
Comment (yihwang-nv) [reply clang/test/dpct/helper_files_ref/include/dpl_extras/algorithm.h:1486]: Same as above

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Pull request merged**